src/pages/home/report/withReportAndPrivateNotesOrNotFound.js

```diff
@@ -0,0 +1,127 @@
+import React, {useEffect, useMemo} from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import lodashGet from 'lodash/get';
+import {withOnyx} from 'react-native-onyx';
+import getComponentDisplayName from '../../../libs/getComponentDisplayName';
+import reportPropTypes from '../../reportPropTypes';
+import withReportOrNotFound from './withReportOrNotFound';
+import * as Report from '../../../libs/actions/Report';
+import NotFoundPage from '../../ErrorPage/NotFoundPage';
+import FullScreenLoadingIndicator from '../../../components/FullscreenLoadingIndicator';
+import {withNetwork} from '../../../components/OnyxProvider';
+import compose from '../../../libs/compose';
+import * as ReportUtils from '../../../libs/ReportUtils';
+import networkPropTypes from '../../../components/networkPropTypes';
+import ONYXKEYS from '../../../ONYXKEYS';
+
+const propTypes = {
+    /** The HOC takes an optional ref as a prop and passes it as a ref to the wrapped component.
+     * That way, if a ref is passed to a component wrapped in the HOC, the ref is a reference to the wrapped component, not the HOC. */
+    forwardedRef: PropTypes.func,
+
+    /** The report currently being looked at */
+    report: reportPropTypes,
+
+    /** Information about the network */
+    network: networkPropTypes.isRequired,
+
+    /** Session of currently logged in user */
+    session: PropTypes.shape({
+        /** Currently logged in user accountID */
+        accountID: PropTypes.number,
+    }),
+
+    route: PropTypes.shape({
+        /** Params from the URL path */
+        params: PropTypes.shape({
+            /** reportID and accountID passed via route: /r/:reportID/notes */
+            reportID: PropTypes.string,
+            accountID: PropTypes.string,
+        }),
+    }).isRequired,
+};
+
+const defaultProps = {
+    forwardedRef: () => {},
+    report: {},
+    session: {
+        accountID: null,
+    },
+};
+
+export default function (WrappedComponent) {
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    function WithReportAndPrivateNotesOrNotFound({forwardedRef, ...props}) {
+        const {route, report, network, session} = props;
+        const accountID = route.params.accountID;
+        const isLoadingPrivateNotes = report.isLoadingPrivateNotes;
```

Comment:
Is `isLoadingPrivateNotes` always set or we should default to false if it doesn't exist?

Reply:
I feel like we can skip undefined case and set `isLoadingPrivateNotes` to true if we're about to, or fetching notes and then render loading indicator

Reply:
done. thanks.